### PR TITLE
Pep8 and other corrections

### PR DESCRIPTION
**Pull request by jpihl**
@mortenvp

### CODE REVIEW

File: test/python/test_http_resolver.py [hunk 1]
```diff
@@ -25,9 +24,11 @@ def create_file(cwd, source, filename):
...
     url_download.download.side_effect = create_file
 
-    resolver = HttpResolver(url_download=url_download, dependency=dependency,
-        source=http_source, cwd=cwd)
+    resolver = HttpResolver(
+        url_download=url_download, dependency=dependency, source=http_source,
+        cwd=cwd)
 
+    # TODO this path is never tested
     path = resolver.resolve()
```
Comment (jpihl) [inline test/python/test_http_resolver.py]: This variable (path) is never tested? Is this a bug?

File: test/python/test_http_resolver.py [hunk 2]
```diff
@@ -53,9 +54,11 @@ def create_file(cwd, source, filename):
 
     url_download.download.side_effect = create_file
 
-    resolver = HttpResolver(url_download=url_download, dependency=dependency,
-        source=http_source, cwd=cwd)
+    resolver = HttpResolver(
+        url_download=url_download, dependency=dependency, source=http_source,
+        cwd=cwd)
 
+    # TODO this path is never tested
```
Comment (jpihl) [inline test/python/test_http_resolver.py]: ditto

File: src/wurf/registry.py [hunk 3]
```diff
@@ -248,14 +249,15 @@ def call():
...
 
                     call = self.registry[provider_name]
                     result = provider_function(**inject_arguments)
                     provider.data[key] = result
                     return result
             else:
+                # TODO: call is never used here, it may be a bug?
                 call = self.registry[provider_name]
                 result = provider_function(**inject_arguments)
                 return result
```
Comment (jpihl) [inline src/wurf/registry.py]: Here the call variable is never used. this must be a bug right?

**Comment by petya2164** [comment]
@jpihl I am not quite sure if we should follow pep8. The "corrected" formatting doesn't look natural at a lot of places. And the 78-char line length is a really bad joke, we should forget that!
In my opinion, using pep8 does not give us better code, conforming to the style feels like unnecessary nitpicking. Maybe we can find a more reasonable standard or live without "enforcing" a style.

**Comment by jpihl** [comment]
I don't want to reinvent the wheel here, pep8 is widely used and there is a lot
of tools that can help with both detecting and fixing pep8 issues. To me that's
the biggest reason to go with pep8.
I did all the changes by hand and I didn't feel that any of the changes were
unnatural. Can you point to a specific place where you feel there is an issue?
Also as stated [here](http://docs.steinwurf.com/coding_style.html#python) we have already decided on using pep8 :)

File: src/wurf/git.py [hunk 4]
```diff
@@ -22,7 +23,7 @@ def cmd_and_log(self, *args, **kwargs):
         :param kwargs: Keyword arguments passed to ctx.cmd_and_log(...)
         """
         args = [self.git_binary] + list(args)
-        return cmd_and_log(args, **kwargs)
+        return self.ctx.cmd_and_log(args, **kwargs)
```
Comment (petya2164) [inline src/wurf/git.py]: cmd_and_log is not used anywhere, so it should be removed instead.

File: src/wurf/http_resolver.py [hunk 5]
```diff
@@ -51,7 +51,8 @@ def resolve(self):
             filename = None
 
         file_path = self.url_download.download(cwd=folder_path,
-            source=self.source, filename=filename)
+                                               source=self.source,
+                                               filename=filename)
```
Comment (petya2164) [inline src/wurf/http_resolver.py]: This is a good example where the pep8-style vertical alignment looks extremely ugly. It is like a toothbrush!
There are lots of examples like this after the reformatting.
@mortenvp and I consistently used the single indentation throughout this project. That looks much better, although it is explicitly forbidden in http://legacy.python.org/dev/peps/pep-0008/#indentation
I think that this kind of vertical alignment should be reverted.

File: test/existing_tag/libbaz/wscript [hunk 6]
```diff
@@ -4,15 +4,19 @@
 APPNAME = 'libbaz'
 VERSION = '1.0.0'
 
+
 def resolve(ctx):
     pass
 
+
```
Comment (petya2164) [inline test/existing_tag/libbaz/wscript]: I think it is totally useless to have 2 empty lines between top-level function. I see that you had to add spaces at million spaces. Let's not waste any effort on that in the future. A single separating line is perfectly fine.

File: src/wurf/git.py (hunk 4 above)
Comment (jpihl) [inline src/wurf/git.py]: before ``cmd_and_log`` was not defined. ``self.ctx.cmd_and_log`` is used in all the other functions in this file so I assumed that it was that ``self.ctx.cmd_and_log`` should be used here as well.

File: src/wurf/http_resolver.py (hunk 5 above)
Comment (jpihl) [inline src/wurf/http_resolver.py]: I removed the toothbrush

File: src/wurf/git.py (hunk 4 above)
Comment (mortenvp) [inline src/wurf/git.py]: I think what @petya2164 means it that we should remove this function all together - I will do that

**Pull request merged**